monai/metrics/meandice.py

```diff
@@ -114,45 +117,124 @@ def compute_dice(
         y_pred: input data to compute, typical segmentation model output.
             It must be one-hot format and first dim is batch, example shape: [16, 3, 32, 32]. The values
             should be binarized.
-        y: ground truth to compute mean dice metric. It must be one-hot format and first dim is batch.
-            The values should be binarized.
+        y: ground truth to compute mean dice metric. `y` can be single-channel class indices or in the one-hot format.
         include_background: whether to skip Dice computation on the first channel of
             the predicted output. Defaults to True.
         ignore_empty: whether to ignore empty ground truth cases during calculation.
             If `True`, NaN value will be set for empty ground truth cases.
             If `False`, 1 will be set if the predictions of empty ground truth cases are also empty.
 
     Returns:
-        Dice scores per batch and per class, (shape [batch_size, num_classes]).
+        Dice scores per batch and per class, (shape: [batch_size, num_classes]).
+
+    """
+    return DiceHelper(  # type: ignore
+        include_background=include_background,
+        reduction=MetricReduction.NONE,
+        get_not_nans=False,
+        softmax=False,
+        ignore_empty=ignore_empty,
+    )(y_pred=y_pred, y=y)
 
-    Raises:
-        ValueError: when `y_pred` and `y` have different shapes.
 
+class DiceHelper:
```

Comment:
What is the rationale behind putting the dice code into a class? This class doesn't really have much state, whatever is passed to its constructor could be arguments for the function like in the original version. The code using this class creates a new instance each time as well which doesn't let us reuse the object's state in future invocations of `__call__`.

Reply:
useful for configuration based workflows to prepare instances cc @myron